server/src/main/java/org/elasticsearch/snapshots/InternalSnapshotsInfoService.java

```diff
@@ -95,7 +98,7 @@ public InternalSnapshotsInfoService(
         this.knownSnapshotShards = Map.of();
         this.unknownSnapshotShards = new LinkedHashSet<>();
         this.failedSnapshotShards = new LinkedHashSet<>();
-        this.queue = new LinkedList<>();
```

Comment:
The array deque is backed by an array, while a LinkedList is chained elements, perhaps there was a reason for this, but either way it should be addressed in a separate PR if you want to make this change.

Reply:
Yup, good idea